src/frontend/posts_page/components/VisibilitySetting.tsx

```diff
@@ -1,48 +1,32 @@
 import React from 'react';
 
-interface IVisibilitySettingState {
-  public: boolean,
-  friendsOnly: boolean,
-  private: boolean
+interface IVisibilitySettingProps {
+  setPublic: any,
+  setFriendsOnly: any,
+  setPrivate: any,
+  visibility: string
 }
 
-class VisibilitySetting extends React.Component<{}, IVisibilitySettingState> {
-  constructor(props: {}) {
+class VisibilitySetting extends React.Component<IVisibilitySettingProps, {}> {
+  constructor(props: IVisibilitySettingProps) {
     super(props);
-    this.state = {
-      public: true,
-      friendsOnly: false,
-      private: false
-    };
   }
 
-  setPublic = () => {
-    this.setState({public: true, friendsOnly: false, private: false});
-  };
-
-  setFriendsOnly = () => {
-    this.setState({public: false, friendsOnly: true, private: false});
-  };
-
-  setPrivate = () => {
-    this.setState({public: false, friendsOnly: false, private: true});
-  };
-
   render() {
     return (
       <div id="visibility-selection">
         <div className="selector-title">Visibility:</div>
         <div className="selector-buttons">
           <label className="selector-button">
-            <input type="radio" checked={this.state.public} onClick={this.setPublic}/>
+            <input type="radio" checked={this.props.visibility === 'public'} onClick={this.props.setPublic}/>
```

Comment:
May be we can chage it to a Enum?

Reply:
True, it visibility features will be important when saving posts into database